python-avd/pyavd/_eos_designs/structured_config/network_services/utils_wan.py

```diff
@@ -37,184 +37,325 @@ def _filtered_wan_vrfs(self: AvdStructuredConfigNetworkServicesProtocol) -> EosD
-            return self.inputs.wan_virtual_topologies.policies
+    @structured_config_contributor
+    def set_wan_policies(self: AvdStructuredConfigNetworkServicesProtocol) -> None:
+        if not self.shared_utils.is_wan_router:
+            return
 
-        policies = self.inputs.wan_virtual_topologies.policies._deepcopy()
-        policies.append(self._default_wan_policy)
-        return policies
+        append_policy = self._append_autovpn_policy if self.inputs.wan_mode == "autovpn" else self._append_cv_pathfinder_policy
 
-    @cached_property
-    def _filtered_wan_policies(self: AvdStructuredConfigNetworkServicesProtocol) -> list:
-        """
-        Loop through all the VRFs defined under `wan_virtual_topologies.vrfs` and returns a list of policies to configure on this device.
+        for vrf in self._filtered_wan_vrfs:
+            if vrf.policy not in self.inputs.wan_virtual_topologies.policies:
+                if vrf.policy == self._default_wan_policy_name and self._default_wan_policy_name not in self.inputs.wan_virtual_topologies.policies:
+                    vrf_policy = self._default_wan_policy
+                else:
+                    msg = (
+                        f"The policy {vrf.policy} applied to vrf {vrf.name} under `wan_virtual_topologies.vrfs` is not "
+                        "defined under `wan_virtual_topologies.policies`."
+                    )
+                    raise AristaAvdInvalidInputsError(msg)
+            else:
+                vrf_policy = self.inputs.wan_virtual_topologies.policies[vrf.policy]
 
-        This returns a structure where every policy contains a list of match statement and a default_match statement if any is required by inputs.
-        Inside each match and default_match statetement, the fully resolved load_balancing policy is present (it guarantees that the load-balance policy
-        is not empty).
+            # TODO: it seems we could set multiple time the same policy here so need to check
+            append_policy(vrf_policy, vrf, control_plane=vrf.name == "default")
 
-        The default VRF is marked as default.
-        """
-        # to track the names already injected
-        filtered_policy_names = []
-        filtered_policies = []
+        # Add Application Traffic Recognition metadata after all policies have been taken care off.
+        self.set_cv_pathfinder_metadata_applications()
 
-        for vrf in self._filtered_wan_vrfs:
-            # Need to handle VRF default differently and lookup for the original policy
-            if vrf.policy not in self._wan_virtual_topologies_policies:
-                msg = (
-                    f"The policy {vrf.policy} applied to vrf {vrf.name} under `wan_virtual_topologies.vrfs` is not "
-                    "defined under `wan_virtual_topologies.policies`."
-                )
-                raise AristaAvdInvalidInputsError(msg)
+    def _append_control_plane_virtual_topology(
```

Comment:
The name here seems misleading, since we are also calling many other functions doing other stuff.